docs/technical/security.md

```diff
@@ -0,0 +1,201 @@
+
+## Development Environment
+
+Whilst the code is still on a developer's machine there are numerous opportunities to avoid and resolve security issues which may arise.
+
+### Development Language Best Practice
+
+Many tools exist which can enforce best practice on the code developed within a project.  These take the form of simple warnings displayed to the developer for common issues with code.  The developer then needs to resolve the issue or add an exception to the codebase.  Exceptions to best practice would be examined in code review and verified before the code is committed.
+
+Many 3rd party tools exist which generate warnings in the developer's IDE for the infractions as they are discovered.
+
+[SonarQube](https://www.sonarqube.org/) is a common source scanning tool and will warn about many features of a project's source code 'health' (dependencies included) and supports a number of languages.
+
+.NET (use by this project at time of writing) has code analysis tools, these should be enabled on all projects.
+
+Other languages have 'linting' tools which should be considered when onboarding a new language for this project.
+
+### Dependency Health
+
+#### .NET Packages
+
+.NET Core references published open source functionality via NuGet Packages.  These packages may have published security issues, Common Vulnerabilities and Exposures (CVE) or GitHub Security Advisory (GHSA) which will require investigation and replacement by the development team to ensure executing code has the best possible security posture.
+
+Tools like GitHub's [Dependabot](https://docs.github.com/en/code-security/dependabot) and [SonarQube](https://www.sonarqube.org/) have the capability to warn developers when these vulnerabilities are discovered.  
+Even the .NET command line tool provides a basic view of dependency issues:
+
+```cmd
+dotnet list package --vulnerable --include-transitive
+```
+
+This [tooling](https://devblogs.microsoft.com/nuget/how-to-scan-nuget-packages-for-security-vulnerabilities/) is very basic however.
+
+At a minimum tooling should be used to warn developers of issues would run in the build pipeline.  Ideally, developers are notified as soon as vulnerabilities are discovered and changes can be made even for components which aren't in active development.
+
+#### Password scanning
+
+All code repos in the Buerokratt platform are public on GitHub.  This means that secrets should be carefully segregated from code.  
+
+In order to prevent developers committing secrets to public repos this project should implement password scanning which would run before changes are committed.
+
+This can be done on a Git commit hook and is documented here:
+
+> See the CentOps repo for an example of this setup.  This should be applied to all repos.
+
+## Built Artefacts
+
+This project aims to package all shipping code in Docker Containers, therefore to remain as secure as possible the project will need to validate they are free of known vulnerabilities.
+

```

Comment:
I know that I might be opening can of worms here but what about container poisoning or any other ways where container name/tag is on some image which is built of unknown source. Signing images after building and verifying the signatures (also third party images) before cluster controller creates new container from the image would help. [One blog post](https://www.securesystems.de/blog/container-image-signatures-in-kubernetes/) about the issue.

Reply:
And one more thought about signing. Should all commits also be signed and verified as first step of any pipeline? Or is it too much? At least 2FA should be mandatory on GitHub when developing open projects.